src/util/graph.h

```diff
@@ -252,6 +252,9 @@ class grapht
   std::vector<node_indext>
   get_reachable(const std::vector<node_indext> &src, bool forwards) const;
 
+  void disconnect_unreachable(node_indext src);
+  void disconnect_unreachable(std::vector<node_indext> src);
```

Comment:
I think this should take a `const std::vector<node_indext> &src` parameter to avoid copying.